File: data/src/main/java/com/github/gmetal/data/entity/MovieEntity.java
package com.github.gmetal.data.entity;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

import java.util.ArrayList;
import java.util.List;

@JsonObject
public class MovieEntity {

    @JsonField
    public Boolean adult;

    @JsonField(name = "backdrop_path")
    public String backdropPath;

    @JsonField(name = "belongs_to_collection")
    public String belongsToCollection;

    @JsonField
    public Integer budget;

    @JsonField
    public List<GenreEntity> genreEntities = new ArrayList<>();

    @JsonField
    public String homepage;

    @JsonField
    public Integer id;

    @JsonField(name = "imdb_id")
    public String imdbId;

    @JsonField(name = "original_language")
    public String originalLanguage;

    @JsonField(name = "original_title")
    public String originalTitle;

    @JsonField
    public String overview;

    @JsonField
    public Double popularity;

    @JsonField(name = "poster_path")
    public String posterPath;

    @JsonField(name = "production_companies")
    public List<ProductionCompanyEntity> productionCompanies = new ArrayList<>();

    @JsonField(name = "production_countries")
    public List<ProductionCountryEntity> productionCountries = new ArrayList<>();

    @JsonField(name = "release_date")
    public String releaseDate;

    @JsonField
    public Integer revenue;

    @JsonField
    public Integer runtime;

    @JsonField(name = "spoken_languages")
    public List<SpokenLanguageEntity> spokenLanguageEntities = new ArrayList<>();

    @JsonField
    public String status;

    @JsonField
    public String tagline;

    @JsonField
    public String title;

    @JsonField
    public Boolean video;

    @JsonField(name = "vote_average")
    public Float voteAverage;

    @JsonField(name = "vote_count")
    public Integer voteCount;
}

File: data/src/main/java/com/github/gmetal/data/entity/MovieEntity.kt
package com.github.gmetal.data.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class MovieEntity {

    var adult: Boolean? = null

    @SerialName("backdrop_path")
    var backdropPath: String? = null

    @SerialName("belongs_to_collection")
    var belongsToCollection: CollectionEntity? = null

    var budget: Int? = null

    var genreEntities: List<GenreEntity> = ArrayList()

    var homepage: String? = null

    var id: Int? = null

    @SerialName("imdb_id")
    var imdbId: String? = null

    @SerialName("original_language")
    var originalLanguage: String? = null

    @SerialName("original_title")
    var originalTitle: String? = null

    var overview: String? = null

    var popularity: Double? = null

    @SerialName("poster_path")
    var posterPath: String? = null

    @SerialName("production_companies")
    var productionCompanies: List<ProductionCompanyEntity> = ArrayList()

    @SerialName("production_countries")
    var productionCountries: List<ProductionCountryEntity> = ArrayList()

    @SerialName("release_date")
    var releaseDate: String? = null

    var revenue: Int? = null

    var runtime: Int? = null

    @SerialName("spoken_languages")
    var spokenLanguageEntities: List<SpokenLanguageEntity> = ArrayList()

    var status: String? = null

    var tagline: String? = null

    var title: String? = null

    var video: Boolean? = null

    @SerialName("vote_average")
    var voteAverage: Float? = null

    @SerialName("vote_count")
    var voteCount: Int? = null
}
